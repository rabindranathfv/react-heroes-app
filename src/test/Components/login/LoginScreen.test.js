import { mount } from 'enzyme';
import { MemoryRouter, Routes, Route } from 'react-router-dom';

import { AuthContext } from './../../../auth/authContext';
import { LoginScreen } from './../../../Components/login/LoginScreen';

// !important when you are using jest.mock remenber use mock<FunctionName> for easily mock without break scope variables
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('LoginScreen', () => { 

    const contextVal = {
        user: {
            logged: true,
            name: 'rabindranath'
        },
        dispatch: jest.fn()
    }

   it('Should render LoginScreen component with user authenticated', () => {
       const wrapper = mount(
        <AuthContext.Provider value={contextVal} > 
            <MemoryRouter initialEntries={['/login']}>
                <Routes>
                    <Route path='/login' element={ <LoginScreen />} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>   
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('button').text().trim()).toBe('Login');
   }); 

   it('Should render LoginScreen with user authenticated, login btn and make login successfuly and redirect to Marvel as based parameter', () => {
    const wrapper = mount(
     <AuthContext.Provider value={contextVal} > 
         <MemoryRouter initialEntries={['/login']}>
                <Routes>
                    <Route path='/login' element={ <LoginScreen />} />
                </Routes>
            </MemoryRouter>
     </AuthContext.Provider>   
    );

    wrapper.find('button').prop('onClick')();

    expect(wrapper.find('button').text().trim()).toBe('Login');
    expect(contextVal.dispatch).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledWith('/marvel', {'replace': true});
    }); 

});