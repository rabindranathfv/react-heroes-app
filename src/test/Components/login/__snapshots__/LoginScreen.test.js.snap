// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Should render LoginScreen component with user authenticated 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
